src/variant.rs

```diff
@@ -122,6 +162,40 @@ impl Variant {
         T::from_variant(self)
     }
 
+    /// Boxes value.
+    #[inline]
+    pub fn new_variant(value: &Variant) -> Self {
+        unsafe {
+            from_glib_full(glib_sys::g_variant_new_variant(value.to_glib_full()))
+        }
+    }
+
+    /// Unboxes self.
+    ///
+    /// Returns `Some` if self contains a `Variant`.
+    #[inline]
+    pub fn get_variant(&self) -> Option<Variant> {
+        unsafe {
+            from_glib_none(glib_sys::g_variant_get_variant(self.to_glib_none().0))
+        }
+    }
+
+    /// Reads a child item out of a container `Variant` instance.
+    ///
+    /// # Panics
+    ///
+    /// * if `self` is not a container type.
+    /// * if given `index` is larger than number of children.
+    pub fn get_child_value(&self, index: usize) -> Variant {
+        assert!(index <= self.n_children());
```

Comment:
```suggestion
        assert!(index < self.n_children());
```

Or not? It's zero-index I assume?